src/main/java/controller/user/AddUserServlet.java

```diff
@@ -0,0 +1,86 @@
+package controller.user;
+
+import dao.impl.UserDaoImpl;
+import factory.UserServiceFactory;
+import model.Role;
+import model.User;
+import org.apache.log4j.Logger;
+import service.UserService;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+@WebServlet("/admin/add/user")
+public class AddUserServlet extends HttpServlet {
+
+    private static final Logger LOGGER = Logger.getLogger(UserDaoImpl.class);
+    private UserService userService = UserServiceFactory.getInstance();
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
```

Comment:
try to simplify this method

Reply:
> try to simplify this method

This method includes checking the fields, should I delete it?